popup.js

```diff
@@ -118,16 +111,20 @@ function savesnaps(){
 
 // Add date and time to text input
 function adddate(){
-
     var months=['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'];
     var dat=new Date();
     var d=dat.getDate();
     var m=months[dat.getMonth()];
     var hr=dat.getHours();
-    if (hr>12){hr=hr-12+":"+dat.getMinutes()+" PM";}
-    else if (hr==12){hr=hr+":"+dat.getMinutes()+" PM";}
-    else {hr=hr+":"+dat.getMinutes()+" AM";}
-    console.log(d);
+    var min=dat.getMinutes();
+    if(d<10){d="0"+d;}
+    if(min<10){min="0"+min;}
+    if (hr>12){hr=hr-12;}
+    else if (hr==12){hr=hr+":"+min+" PM";}
+    else {hr=hr+":"+min+" AM";}
+    if(hr<12){hr= "0"+hr+":"+min+" PM";}
```

Comment:
You're still comparing string with integer in last line which will always turn out to be false. Your code will not work for time between 12 AM and 12 PM. Suppose time is "07:45 AM". It will still show as "7:45 AM", not "07:45 AM".